ps_emailsubscription.php

```diff
@@ -792,14 +795,50 @@ public function hookActionCustomerAccountAdd($params)
     {
         //if e-mail of the created user address has already been added to the newsletter through the ps_emailsubscription module,
         //we delete it from ps_emailsubscription table to prevent duplicates
+        if (empty($params['newCustomer'])) {
+            return;
+        }  
         $id_shop = $params['newCustomer']->id_shop;
         $email = $params['newCustomer']->email;
         if (Validate::isEmail($email)) {
             return (bool) Db::getInstance()->execute('DELETE FROM '._DB_PREFIX_.'emailsubscription WHERE id_shop='.(int) $id_shop.' AND email=\''.pSQL($email)."'");
         }
+        
+        if ($params['newCustomer']->newsletter) {
```

Comment:
Can you please store the object:
```php
/** @var Customer $customer */
$customer = $params['newCustomer'];
```

This will prevent the hard to ready notation (`$params['newCustomer']->newsletter` mixing arrays and objects)